agent/config/runtime_test.go

```diff
@@ -4997,1495 +4986,28 @@ func TestFullConfig(t *testing.T) {
-				key = "j9lF1Tve"
-				handler = "90N7S4LN"
-			}, {
-				type = "keyprefix"
-				datacenter = "fYrl3F5d"
-				key = "sl3Dffu7"
-				args = ["dltjDJ2a", "flEa7C2d"]
-			}]
-		`}
-
-	tail := map[string][]Source{
-		"json": {
-			FileSource{
-				Name:   "tail.non-user.json",
-				Format: "json",
-				Data: `
-				{
-					"acl_disabled_ttl": "957s",
-					"acl" : {
-						"disabled_ttl" : "957s"
-					},
-					"ae_interval": "10003s",
-					"check_deregister_interval_min": "27870s",
-					"check_reap_interval": "10662s",
-					"discovery_max_stale": "5s",
-					"segment_limit": 24705,
-					"segment_name_limit": 27046,
-					"sync_coordinate_interval_min": "27983s",
-					"sync_coordinate_rate_target": 137.81
-				}`,
-			},
-			FileSource{
-				Name:   "tail.consul.json",
-				Format: "json",
-				Data: `
-				{
-					"consul": {
-						"coordinate": {
-							"update_batch_size": 9244,
-							"update_max_batches": 15164,
-							"update_period": "25093s"
-						},
-						"raft": {
-							"election_timeout": "31947s",
-							"heartbeat_timeout": "25699s",
-							"leader_lease_timeout": "15351s"
-						},
-						"server": {
-							"health_interval": "17455s"
-						}
-					}
-				}`,
-			},
-		},
-		"hcl": {
-			FileSource{
-				Name:   "tail.non-user.hcl",
-				Format: "hcl",
-				Data: `
-					acl_disabled_ttl = "957s"
-					acl = {
-						disabled_ttl = "957s"
-					}
-					ae_interval = "10003s"
-					check_deregister_interval_min = "27870s"
-					check_reap_interval = "10662s"
-					discovery_max_stale = "5s"
-					segment_limit = 24705
-					segment_name_limit = 27046
-					sync_coordinate_interval_min = "27983s"
-					sync_coordinate_rate_target = 137.81
-				`,
-			},
-			FileSource{
-				Name:   "tail.consul.hcl",
-				Format: "hcl",
-				Data: `
-					consul = {
-						coordinate = {
-							update_batch_size = 9244
-							update_max_batches = 15164
-							update_period = "25093s"
-						}
-						raft = {
-							election_timeout = "31947s"
-							heartbeat_timeout = "25699s"
-							leader_lease_timeout = "15351s"
-						}
-						server = {
-							health_interval = "17455s"
-						}
-					}
-				`,
-			},
-		},
-	}
-
-	expected := RuntimeConfig{
+	expected := &RuntimeConfig{
 		// non-user configurable values
-		ACLDisabledTTL:             957 * time.Second,
-		AEInterval:                 10003 * time.Second,
-		CheckDeregisterIntervalMin: 27870 * time.Second,
-		CheckReapInterval:          10662 * time.Second,
-		SegmentLimit:               24705,
-		SegmentNameLimit:           27046,
-		SyncCoordinateIntervalMin:  27983 * time.Second,
-		SyncCoordinateRateTarget:   137.81,
+		ACLDisabledTTL:             120 * time.Second,
+		AEInterval:                 time.Minute,
+		CheckDeregisterIntervalMin: time.Minute,
+		CheckReapInterval:          30 * time.Second,
+		SegmentLimit:               64,
+		SegmentNameLimit:           64,
+		SyncCoordinateIntervalMin:  15 * time.Second,
+		SyncCoordinateRateTarget:   64,
```

Comment:
These values changed because the `tail` overrides were removed. These overrides only existed so that `nonZero` could attempt to enforce unique values for all config. However that enforcement never seemed to work (see the TODO(fs) and the new FIXME below).

Removing them makes the test scenario more realistic because normally users can not set these values. In the future we should remove these entirely from `RuntimeConfig`.